src/bundle.c

```diff
@@ -1641,6 +1641,147 @@ gboolean check_bundle_payload(RaucBundle *bundle, GError **error)
 	return res;
 }
 
+gboolean replace_signature(RaucBundle *bundle, const gchar *insig, const gchar *outpath, CheckBundleParams params, GError **error)
+{
+	g_autoptr(RaucManifest) manifest = NULL;
+	g_autoptr(RaucBundle) outbundle = NULL;
+	g_autoptr(GFile) bundleoutfile = NULL;
+	GFileIOStream* bundlestream = NULL;
+	GOutputStream* bundleoutstream = NULL;
+	g_autoptr(GBytes) sig = NULL;
+	gchar* keyringpath = NULL;
+	gchar* keyringdirectory = NULL;
+	GError *ierror = NULL;
+	gboolean res = FALSE;
+	gsize sigsize;
+
+	g_return_val_if_fail(bundle != NULL, FALSE);
+	g_return_val_if_fail(outpath != NULL, FALSE);
+	g_return_val_if_fail(insig != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	if (!(params & CHECK_BUNDLE_NO_VERIFY) && r_context()->signing_keyringpath) {
+		keyringpath = r_context()->config->keyring_path;
+		keyringdirectory = r_context()->config->keyring_directory;
+		r_context()->config->keyring_path = r_context()->signing_keyringpath;
+		r_context()->config->keyring_directory = NULL;
+	}
+
+	r_context_begin_step("replace_signature", "Replacing bundle signature", 5);
```

Comment:
Setting progress step should not be required here as this will not be called in an installation context.

Reply:
We have the same in extract_signature and extract_bundle.

Reply:
Not that this is better. :smirk: 
I already have a local patch for removing them, thus add it so I can remove it...

Reply:
It also helps with verification that we follow the error exit paths. :)